test/tesla/middleware/telemetry_test.exs

```diff
@@ -23,8 +23,12 @@ defmodule Tesla.Middleware.TelemetryTest do
     :telemetry.attach(
       "telemetry_test",
       [:tesla, :request],
-      fn [:tesla, :request], response, meta, _config ->
-        send(self(), {:ok_called, is_integer(response), meta})
+      fn
+        [:tesla, :request], %{value: time}, meta, _config ->
+          send(self(), {:ok_called, is_integer(time), meta})
+
+        [:tesla, :request], response, meta, _config ->
```

Comment:
Why not add a guard matcher here to check if `response` is an integer?

Reply:
Yeah, the guard is more concise and precise here. 
I want to know if it is proper to raise an assertion error from guard/pattern match in elixir for unit test. As I come from Java, most of my unit tests I have written indicate the assertion in ```assertThat()```. 
Anyway thank you for the comment

Reply:
Totally, no problem! Thank _you_ for working on this!